tests/e2e/performance/test_pvc_snapshot_performance.py

```diff
@@ -0,0 +1,233 @@
+            )
+
+            # Wait for fio to finish
+            fio_result = self.pod_obj.get_fio_results()
+            err_count = fio_result.get('jobs')[0].get('error')
+            assert err_count == 0, (
+                f"IO error on pod {self.pod_obj.name}. "
+                f"FIO result: {fio_result}"
+            )
+            log.info('IO on the PVC Finished')
+
+            log.info(f"Verified IO on pod {self.pod_obj.name}.")
+
+            # Verify presence of the file
+            file_path = pod.get_file_path(self.pod_obj, file_name)
+            log.info(f"Actual file path on the pod {file_path}")
+            assert pod.check_file_existence(self.pod_obj, file_path), (
+                f"File {file_name} doesn't exist"
+            )
+            log.info(f"File {file_name} exists in {self.pod_obj.name}")
+
+            # Calculate md5sum
+            orig_md5_sum = pod.cal_md5sum(self.pod_obj, file_name)
+
+            # Take a snapshot
+            snap_name = self.pvc_obj.name.replace(
+                'pvc-test', f'snapshot-test{test_num}'
+            )
+            log.info(f'Taking snapshot of the PVC {snap_name}')
+            snap_obj = pvc.create_pvc_snapshot(
+                self.pvc_obj.name,
+                snap_yaml,
+                snap_name,
+                helpers.default_volumesnapshotclass(self.interface).name,
+            )
+            snap_obj.ocp.wait_for_resource(
+                condition='true', resource_name=snap_obj.name,
+                column=constants.STATUS_READYTOUSE, timeout=60
+            )
+            teardown_factory(snap_obj)
+            create_times.append(
+                helpers.measure_snapshot_creation_time(snap_obj.name)
+            )
+
+            # Same Storage class of the original PVC
+            sc_name = self.pvc_obj.backed_sc
+
+            # Size should be same as of the original PVC
+            pvc_size = str(self.pvc_obj.size) + "Gi"
+
+            # Create pvc out of the snapshot
+            # Both, the snapshot and the restore PVC should be in same namespace
+
+            log.info('Restoring from the Snapshot')
+            restore_pvc_name = self.pvc_obj.name.replace(
+                'pvc-test', f'restore-pvc{test_num}'
+            )
+            restore_pvc_yaml = constants.CSI_RBD_PVC_RESTORE_YAML
+            if self.interface == constants.CEPHFILESYSTEM:
+                restore_pvc_yaml = constants.CSI_CEPHFS_PVC_RESTORE_YAML
+
+            log.info('Resorting the PVC from Snapshot')
+            restore_pvc_obj = pvc.create_restore_pvc(
+                sc_name=sc_name, snap_name=snap_obj.name,
+                namespace=snap_obj.namespace, size=pvc_size,
+                pvc_name=restore_pvc_name,
+                restore_pvc_yaml=restore_pvc_yaml
+            )
+            helpers.wait_for_resource_state(
+                restore_pvc_obj,
+                constants.STATUS_BOUND
+            )
+            restore_pvc_obj.reload()
```

Comment:
PVC access mode or VolumeMode can be changed when you are restoring the PVC (this is supported in upstream)

Reply:
Can the volume mode be changed in downstream ?

Reply:
not sure about downstream (upstream it's supported)

Reply:
do we want to to test with different access mode ?

Reply:
@Avilir IMO, we don't have to do it as part of performance test

Reply:
> @Avilir IMO, we don't have to do it as part of performance test

My thought as well